packages/fuels-types/src/transaction_builders.rs

```diff
@@ -482,3 +485,25 @@ pub fn create_message_predicate(
         predicate_data,
     )
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn storage_slots_are_sorted_when_set() {
+        let unsorted_storage_slots = [2, 1].map(given_a_storage_slot).to_vec();
+        let sorted_storage_slots = [1, 2].map(given_a_storage_slot).to_vec();
+
+        let builder = CreateTransactionBuilder::default().set_storage_slots(unsorted_storage_slots);
+
+        assert_eq!(builder.storage_slots, sorted_storage_slots);
```

Comment:
Taking self as `mut self` doesn't mean that the original will be mutated. 

Rather it means that we're taking ownership of whatever is passed in and then **after** it is ours we'll mutate it a bit as well.

For this to be a side-effect it would have to be `&mut self` and have the owner remain outside so he can witness the sorting side-effect.

Reply:
Oh yeah nevermind got it